src/librustdoc/json/conversions.rs

```diff
@@ -240,7 +240,8 @@ fn from_clean_item(item: clean::Item, tcx: TyCtxt<'_>) -> ItemEnum {
 
 impl FromWithTcx<clean::Struct> for Struct {
     fn from_tcx(struct_: clean::Struct, tcx: TyCtxt<'_>) -> Self {
-        let clean::Struct { struct_type, generics, fields, fields_stripped } = struct_;
+        let fields_stripped = struct_.has_stripped_fields();
```

Comment:
Can we remove these _stripped fields from the JSON types too?

Reply:
`fields_stripped` is computed from `fields`, though, so all the necessary data should be there.

Stripped fields should be represented as `StrippedItem(...)`, not by just removing them.

Reply:
> Stripped fields are represented as StrippedItem(...), not by just removing them. If they are ever removed, that's a bug that should be fixed.

Do you mean the current way of documenting private fields is wrong? If so, why

Reply:
Sorry, I'm not sure what you're asking. This PR is based on the property that fields (and other items) are "stripped" by wrapping them with `StrippedItem`. If that's not always the case, then this PR is broken, and we should fix it so that fields are never straight-up removed.

Reply:
Yeah, I agree with all that. But when the JSON encounters a `StrippedItem`, it isn't outputted to the JSON, which I think is why we're talking passed eachother. However in the JSON type for a struct, we need to have the `fields_stripped` bool, for the reason described above, so we cannot remove it. Or were you orrigionaly asking something else?

Reply:
Ah, I understand now. I didn't realize that `StrippedItem`s were removed from the JSON output.